### ISSUE TITLE
dotnet fm rollback all does not work

### ISSUE THREAD

**Issue by gandarez** [fluentmigrator/fluentmigrator#1123]
Running the command below gives me an error

```bash
dotnet fm rollback all -p mysql -c "Server=localhost;Database=Db;Uid=root;Pwd=xpto;" -a MyAssmbly.DbMigration\bin\Debug\netcoreapp2.1\MyAssmbly.DbMigration.dll --tag Testing
```

```
Unrecognized option '-p'
```

Am I missing something?

**Comment by jzabroski** [comment]
Looks like a bug

**Comment by jzabroski** [comment]
@gandarez Just for triage, are you on the latest version?

**Comment by gandarez** [comment]
I think so - `Tool 'fluentmigrator.dotnet.cli' (version '3.2.1') was successfully installed.`

**Comment by gandarez** [comment]
Also for other commands like `migrate up`, `migrate down`, `rollback to` and `rollback by`

![Capture](https://user-images.githubusercontent.com/782854/70151931-46ed4f00-167a-11ea-842a-8410e7bf5cb1.PNG)

**Comment by gandarez** [comment]
I asked them about chaining SubCommands and the answer is `Yes`

https://github.com/natemcmaster/CommandLineUtils/issues/310

**Comment by jzabroski** [comment]
I think the way these commands are written probably needs refactoring.  For my taste, it doesn't make a lot of sense that we effectively have three pieces of glue code for the old-school CLI, the dotnet-fm tool, and MSBuild task.  It makes consistency very hard to ensure.

**Comment by gandarez** [comment]
You mean istead of calling `dotnet fm rollback all` we should call `dotnet fm rollback-all` or `dotnet fm rollback-to` and so on... maybe?

**Comment by jzabroski** [comment]
No, I mean we are using Mono Options in one place, Nate's Command Line Utils in another, etc.  It complicates things by having three different ways to enter the Runner

**Comment by fubar-coder** [comment]
A rollback command doesn't support assemblies and/or tags. This seems like an oversight on my side. The `-p` and `-c` options are connection options and come directly after the `rollback` command.

Example:

```text
dotnet fm rollback -p mysql -c "Server=localhost;Database=Db;Uid=root;Pwd=xpto;" all
```

**Comment by gandarez** [comment]
> A rollback command doesn't support assemblies and/or tags. This seems like an oversight on my side. The `-p` and `-c` options are connection options and come directly after the `rollback` command.
> 
> Example:
> 
> ```
> dotnet fm rollback -p mysql -c "Server=localhost;Database=Db;Uid=root;Pwd=xpto;" all
> ```

Doesn't work

```
The --assembly field is required.
```

**Comment by jzabroski** [comment]
@gandarez Thanks for your help and detailed, simple repros.  Your patience is appreciated.

**Comment by gandarez** [comment]
Is there any easy way of debugging the `Dotnet.Cli` instead of packing and installing locally?

**Comment by jzabroski** [comment]
It's a .net core project so launch.json for VS Code or launchSettings.json
for VS2019.

Configure an attach to process similar to AttachToAnything vs gallery
plugin.

**Comment by fubar-coder** [comment]
I (maybe) misread the source code. Does it work when you specify the other tags too?

**Comment by gandarez** [comment]
Those commands are not working

- rollback by
- rollback to
- rollback all
- migrate up
- migrate down

**Comment by jzabroski** [comment]
@gandarez I'll try to write up some more explicit instructions in the README.md for how people can debug these tasks.  Effectively, the trick is to use Task.ContinueWith and the new ability to specify `async Main` program entry points in C#. I use this trick to debug my PowerShell cmdlets; In PowerShell, effectively I configure my launchSettings.json to start PowerShell, and I run a `Start-Debugger` cmdlet I wrote to attach Visual Studio to the PowerShell window, which in turn attaches to the presently loaded PowerShell modules; a similar trick can be done with FluentMigrator, I think. Thanks for all your feedback.

**Comment by gandarez** [comment]
I found the problem. I fixed most of them and there's one last bug on `rollback` that needs attention. As soon as I finish the tests I do a PR.

**Comment by jzabroski** [comment]
For a :ghost: You sure do show up.

**Comment by gandarez** [comment]
https://github.com/fluentmigrator/fluentmigrator/pull/1130 done

**Comment by jzabroski** [comment]
@gandarez Terribly sorry I've made you wait so long.  Life got so crazy this year with COVID-19.  I'm planning to ship this as part of 3.2.7 this week.  Waiting to hear back from Nate McMaster on what the proper use of his library is on that getter/setter thing.  To say it's ridiculous that this PR has waited so long because of that minor issue is an understatement.  I hope you're still using FluentMigrator and will be happy to see this resolved.

**Comment by jzabroski** [comment]
Nate's response:

> There is no best practice. The setter is optional. Invisible setters are supported because some may want to make their properties immutable
